File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/enums/EnumSODAExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

using System;
using com.db4o;
using com.db4o.query;

namespace com.db4odoc.f1.enums
{
	class EnumSODAExample 
	{
		public static ObjectSet FindOpenDoors(ObjectContainer container) 
		{
			Query query = container.Query();
			query.Constrain(typeof(Door));
			query.Descend("_state").Constrain(DoorState.Open);
			return query.Execute();
		}
	}
}

File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/enums/EnumSODAExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports com.db4o
Imports com.db4o.query

Namespace com.db4odoc.f1.Enums
    Public Class EnumSODAExample
        Public Shared Function FindOpenDoors(ByVal container As ObjectContainer) As ObjectSet
            Dim query As Query = container.Query()
            query.Constrain(GetType(Door))
            query.Descend("_state").Constrain(DoorState.Open)
            Return query.Execute()
        End Function
    End Class
End Namespace
	